### PR TITLE
importing and Jupyter

### PR DESCRIPTION
**Pull request by jedbrown**
The `Pkg.add("./")` fails when I try it, and doesn't seem to be necessary.

### CODE REVIEW

File: examples/jupyter/demo002_diffusion.ipynb [hunk 1]
```diff
@@ -15,9 +17,9 @@
    "source": [
     "# dependencies\n",
     "using Pkg\n",
-    "Pkg.activate(\"./\")\n",
+    "Pkg.activate(\".\")\n",
     "Pkg.instantiate()\n",
-    "Pkg.develop(path=\"./../../../LFAToolkit.jl\")\n",
+    "Pkg.develop(path=\"../..\")\n",
```
Comment (jedbrown) [inline examples/jupyter/demo002_diffusion.ipynb:22]: I think it's better to not require that the clone be named exactly this. The simpler form works for me.

File: examples/jupyter/demo002_diffusion.ipynb [hunk 2]
```diff
@@ -65,12 +67,10 @@
...
    "source": [
     "# compute full operator symbols\n",
-    "numbersteps = 250\n",
-    "maxeigenvalue = 0\n",
+    "numbersteps = 80\n",
     "θ_min = -π/2\n",
     "θ_max = 3π/2\n",
-    "θ_step = 2π/(numbersteps-1)\n",
-    "θ_range = θ_min:θ_step:θ_max\n",
+    "θ_range = LinRange(θ_min, θ_max, numbersteps)\n",
```
Comment (jedbrown) [inline examples/jupyter/demo002_diffusion.ipynb:73]: More idiomatic.

File: examples/jupyter/demo002_diffusion.ipynb [hunk 3]
```diff
@@ -82,12 +82,11 @@
     "        θ = [θ_range[i]]\n",
     "        A = computesymbols(diffusion, θ)\n",
     "        eigenvalues = [abs(val) for val in eigvals(A)]\n",
-    "        maxeigenvalues[i] = max(eigenvalues...)\n",
-    "        maxeigenvalue = max(maxeigenvalue, maxeigenvalues[i])\n",
+    "        maxeigenvalues[i] = maximum(eigenvalues)\n",
```
Comment (jedbrown) [inline examples/jupyter/demo002_diffusion.ipynb:85]: I don't know if this is a performance issue (depends whether unpacking into many arguments is eliminated), but `maximum` operates over arrays.

File: examples/jupyter/demo002_diffusion.ipynb [hunk 4]
```diff
@@ -98,9 +97,9 @@
     "        ylabel=\"spectral radius\",\n",
     "        linewidth=3,\n",
     "        legend=:none,\n",
-    "        title=\"Mass Operator Symbol\"\n",
+    "        title=\"Diffusion Operator Symbol\"\n",
```
Comment (jedbrown) [inline examples/jupyter/demo002_diffusion.ipynb:100]: Oops.

**Review by jedbrown — COMMENTED** [review]
As a feature, I'd really like to avoid stacking these lines on top of each other and especially, taking the max. The 1D plot we have now is characterizing small values of theta by the high frequency modes that are aliased onto it, not by the actual low frequencies.

image

**Comment by jeremylt** [comment]
These look good and I can propagate them to the other examples this afternoon. I think the `Pkg.add("./")` used to be required but is no longer.

**Comment by jedbrown** [comment]
Cool, thanks. Should we make a library utility that computes eigenvalues (optionally also eigenvectors) across a grid of theta? What about creating and annotating plots? Would save a lot of repetition, though it's more magic.

**Comment by jeremylt** [comment]
I think making eigenvalues/vectors across a range of theta is a good idea. My initial impulse is to let people make their own plots though.

**Review by jeremylt — APPROVED** [review]

**Comment by jeremylt** [comment]
I'd like to do the eigenvalues/vectors utility in a separate PR, so if you don't have other Jupyter changes planned, I think this branch can be merged.

**Pull request merged**